pkg/testcase/secretsencrypt.go

```diff
@@ -0,0 +1,170 @@
+package testcase
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestSecretsEncryption() {
+	nodes, errGetNodes := shared.GetNodesByRoles("etcd", "control-plane")
+	Expect(nodes).NotTo(BeEmpty())
+	Expect(errGetNodes).NotTo(HaveOccurred(), "error getting etcd/control-plane nodes")
+
+	product, err := shared.Product()
+	Expect(err).NotTo(HaveOccurred(), "error getting product from config")
+
+	ips := getNodeIps(nodes)
+
+	errSecret := shared.CreateSecret("secret1", "default")
+	Expect(errSecret).NotTo(HaveOccurred(), "error creating secret")
+	shared.LogLevel("INFO", "TEST: 'CLASSIC' Secrets Encryption method")
+	index := len(nodes) - 1
+	secretsEncryptOps("prepare", product, nodes[index].ExternalIP, ips)
+	secretsEncryptOps("rotate", product, nodes[index].ExternalIP, ips)
+	secretsEncryptOps("reencrypt", product, nodes[index].ExternalIP, ips)
+
+	if strings.Contains(os.Getenv("TEST_TYPE"), "both") {
+		shared.LogLevel("INFO", "TEST: 'NEW' Secrets Encryption method")
+		secretsEncryptOps("rotate-keys", product, nodes[index].ExternalIP, ips)
+	}
+}
+
+func secretsEncryptOps(action, product, cpIp string, ips []string) {
+	shared.LogLevel("INFO", fmt.Sprintf("TEST: Secrets-Encryption: %s", action))
+	_, errStatusB4 := shared.SecretEncryptOps("status", cpIp, product)
+	Expect(errStatusB4).NotTo(HaveOccurred(), "error getting secret-encryption status before action")
+
+	stdOutput, err := shared.SecretEncryptOps(action, cpIp, product)
+	Expect(err).NotTo(HaveOccurred(), "error: secret-encryption: "+action)
+	verifyStdOut(action, stdOutput)
+	if (action == "reencrypt") || (action == "rotate-keys") {
+		shared.LogLevel("DEBUG", "reencrypt op needs some time to complete - Sleep for 20 seconds before service restarts")
```

Comment:
why this message is DEBUG and not INFO , on your pov ?

Reply:
Well, if someone doesnt want this seen eventually, when we have the option to drop debug log lines.. wanted to leave it in debug instead of info - in case folks may consider such messages noisy and not necessary.

Reply:
got it tks